network.go

```diff
@@ -0,0 +1,10 @@
+package caddy
+
+import (
+	"net/http"
+	"net/url"
+)
+
+type ProxyFuncProducer interface {
```

Comment:
How necessary is this for it to have its own file? Can we move this into an existing file?

Reply:
It isn't really necessary. I was setting context. I've moved it to an existing file.